### PR TITLE
Update Waterline to 1.1.0

### PR DESCRIPTION
**Pull request by kevinburkeshyp**
The primary change here is to remove the 'alter' migration strategy from
Waterline. In addition, we remove a lot of code related to dynamic finders and
association finders, both of which we don't use.

Diff is here: https://github.com/shyp/waterline/compare/v1.0.1...v1.1.0

In addition we upgraded waterline-schema to 0.2.0, to remove 'alter' as the
default migration strategy.

https://github.com/shyp/waterline-schema/compare/shyp-master-0.1.17...v0.2.0

### CODE REVIEW

**Comment by caseycrites** [comment]
Looks like at least one test is still relying on `alter` being there.

**Comment by kevinburkeshyp** [comment]
Yeahhhhh... working on it

**Comment by kevinburkeshyp** [comment]
Back to green @caseycrites !

File: package.json [hunk 1]
```diff
@@ -70,6 +70,7 @@
     "istanbul": "*",
     "microtime": "~0.6.0",
     "mocha": "2.3.4",
+    "npm": "2",
```
Comment (caseycrites) [inline package.json:73]: Is this related?

Comment (kevinburkeshyp) [inline package.json:73]: I can pull it into a different branch... without this clingwrap tries to install npm 3, which we don't want

Comment (caseycrites) [inline package.json:73]: Yeah, let's pull it into another branch.

**Comment by caseycrites** [comment]
:shaved_ice: after the npm comment.
